src/xdg-autostart-generator/xdg-autostart-service.c

```diff
@@ -0,0 +1,655 @@
+                                 e);
+                        if (!condition_string)
+                                return log_oom();
+                } else {
+                        /* Unsupported or something is wrong, we should never run this service */
+                        log_warning("Not generating service for XDG autostart %s with unsupported AutostartCondition", service->name);
+                        return 0;
+                }
+        }
+
+        path_escaped = specifier_escape(service->path);
+        if (!path_escaped)
+                return log_oom();
+
+        unit = prefix_roota(dest, service->name);
+
+        f = fopen(unit, "wxe");
+        if (!f)
+                return log_error_errno(errno, "Failed to create unit file %s: %m", unit);
+
+        fprintf(f,
+                "# Automatically generated by systemd-xdg-autostart-generator\n\n"
+                "[Unit]\n"
+                "Documentation=man:systemd-xdg-autostart-generator(8)\n"
+                "SourcePath=%s\n"
+                /* Disallow manual starting to make Conflicts= safer. */
+                "RefuseManualStart=yes\n"
+                "PartOf=graphical-session.target\n"
+                "Requisite=xdg-desktop-autostart.target\n",
+                path_escaped);
+
+        if (service->description) {
+                _cleanup_free_ char *t = NULL;
+
+                t = specifier_escape(service->description);
+                if (!t)
+                        return log_oom();
+
+                fprintf(f, "Description=%s\n", t);
+        }
+
+        if (condition_string)
+                fprintf(f, "\n%s\n", condition_string);
+
+        /* GNOME autostart phase emulation
+         *
+         * TODO: We will need to deal with other desktop environments to some
+         *       extend for compatibility.
+         * TODO: The targets here used with the GNOME logic need fixing. */
+        if (STRPTR_IN_SET(service->gnome_autostart_phase,
+                               "PreDisplayServer",
+                               "DisplayServer",
+                               "Initialization"))
+                fprintf(f,
+                        "After=graphical-session-pre.target\n"
+                        "Before=xdg-desktop-autostart.target\n");
+        else if (STRPTR_IN_SET(service->gnome_autostart_phase,
+                               "WindowManager",
+                               "Panel",
+                               "Desktop"))
+                fprintf(f,
+                        "After=xdg-desktop-autostart.target\n"
+                        "Before=graphical-session.target\n");
+       else
+                fprintf(f,
+                        "After=graphical-session.target\n");
+
+        fprintf(f,
+                "\n[Service]\n"
+                "Type=simple\n"
+                "ExecStart=:%s\n"
+                "Restart=no\n"
+                "TimeoutSec=5sec\n"
```

Comment:
doesn't matter, but the canonical suffix spelling for seconds is "s", i.e. maybe change this to 5s